internal/pkg/store/project.go

```diff
@@ -25,9 +26,29 @@ func (s *Store) CreateProject(project *archer.Project) error {
 	}
 
 	if project.Domain != "" {
-		in := &route53domains.GetDomainDetailInput{DomainName: aws.String(project.Domain)}
-		if _, err := s.domainsClient.GetDomainDetail(in); err != nil {
-			return fmt.Errorf("get domain details for %s: %w", project.Domain, err)
+		domainExist := false
+		in := &route53.ListHostedZonesByNameInput{DNSName: aws.String(project.Domain)}
+		resp, err := s.route53Svc.ListHostedZonesByName(in)
+		if err != nil {
+			return fmt.Errorf("list hosted zone for %s: %w", project.Domain, err)
+		}
+		if hostedZoneExists(resp.HostedZones, project.Domain) {
+			domainExist = true
+		} else {
+			for aws.BoolValue(resp.IsTruncated) {
+				in = &route53.ListHostedZonesByNameInput{DNSName: resp.NextDNSName, HostedZoneId: resp.NextHostedZoneId}
+				resp, err = s.route53Svc.ListHostedZonesByName(in)
+				if err != nil {
+					return fmt.Errorf("list hosted zone for %s: %w", project.Domain, err)
+				}
+				if hostedZoneExists(resp.HostedZones, project.Domain) {
+					domainExist = true
+					break
+				}
+			}
+		}
+		if !domainExist {
+			return fmt.Errorf("no hosted zone found for %s", project.Domain)
```

Comment:
Should we move this logic to `route53/route53.go`? Moving it to the route53 pkg would allow a different consumer to leverage the `HostedZoneExists` method.

Reply:
nit: As a fun exercise :D, can you think of an alternative to reduce nesting with the [do-while](https://yourbasic.org/golang/do-while-loop/) style?

Reply:
👍